### PR TITLE
cmake: Fix check for vasprintf

### PR DESCRIPTION
**Pull request by diizzyy**
This fixes the check by changing stdio to stdio.h

### CODE REVIEW

**Comment by diizzyy** [comment]
@Conan-Kudo 
Can you have a look at this one?

File: CMakeLists.txt [hunk 1]
```diff
@@ -59,7 +59,7 @@ chkdef(setreuid unistd.h)
 chkdef(setuid unistd.h)
 chkdef(stpcpy string.h)
 chkdef(strerror string)
```
Comment (Conan-Kudo) [inline CMakeLists.txt:61]: so then this should be `string.h` too, right?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (Conan-Kudo) [reply CMakeLists.txt:61]: Nope.

Comment (diizzyy) [reply CMakeLists.txt:61]: Looking at config.h from autotools these checks / definitions seems to be missing

MacOS functions
HAVE_CFLOCALECOPYCURRENT
HAVE_CFPREFERENCESCOPYAPPVALUE

Functions
HAVE_SECURE_GETENV
HAVE___SECURE_GETENV

Header files
HAVE_DLFCN_H
HAVE_INTTYPES_H
HAVE_MCHECK_H
HAVE_MINIX_CONFIG_H
HAVE_STRINGS_H
HAVE_SYS_STAT_H
HAVE_SYS_TYPES_H

Comment (pmatilai) [reply CMakeLists.txt:61]: Many of those aren't used in the codebase at all, so lets be careful not to add back remants from something that was in autoconf 20 years ago :laughing:

**Review by Conan-Kudo — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (diizzyy) [reply CMakeLists.txt:61]: That's very likely true, I just compared it to current CMake build :-)

It compiles with this PR change test also checks out fine

Comment (pmatilai) [reply CMakeLists.txt:61]: Yep, these need to be reviewed one by one. 
Anyway, thanks for the patch, this one is obvious enough :D

**Pull request merged**